### PR TITLE
72144 endpoint for getting unique filter values for modes

### PR DESCRIPTION
**Pull request by llykkelarsen**
Endpoint for unique filter values: Modes
Tests

### CODE REVIEW

File: src/tests/Equinor.Procosys.Preservation.Query.Tests/GetUniqueTagModes/GetUniqueTagModesQueryHandlerTests.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+                var result = await dut.Handle(_queryForProject1, default);
+                Assert.AreEqual(1, result.Data.Count);
+                Assert.IsTrue(result.Data.Any(rt => rt.Title == _testDataSet.Mode1.Title));
+
+                result = await dut.Handle(new GetUniqueTagModesQuery(_testDataSet.Project2.Name), default);
+                Assert.AreEqual(1, result.Data.Count);
+                Assert.IsTrue(result.Data.Any(rt => rt.Title == _testDataSet.Mode1.Title));
+
+                result = await dut.Handle(new GetUniqueTagModesQuery("Unknown"), default);
```
Comment (TordJoranger) [inline src/tests/Equinor.Procosys.Preservation.Query.Tests/GetUniqueTagModes/GetUniqueTagModesQueryHandlerTests.cs]: A sepperate test for this I think?

File: src/Equinor.Procosys.Preservation.Query/GetUniqueTagModes/GetUniqueTagModesQueryHandler.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+        public GetUniqueTagModesQueryHandler(IReadOnlyContext context) => _context = context;
+
+        public async Task<Result<List<ModeDto>>> Handle(GetUniqueTagModesQuery request,
+            CancellationToken cancellationToken)
+        {
+            var modes = await
+                (from mode in _context.QuerySet<Mode>()
+                    join step in _context.QuerySet<Step>()
+                        on mode.Id equals EF.Property<int>(step, "ModeId")
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Query/GetUniqueTagModes/GetUniqueTagModesQueryHandler.cs]: on mode.Id equals step.ModeId (ModeId is not a shadow property ... it exists)

File: src/tests/Equinor.Procosys.Preservation.Query.Tests/GetUniqueTagModes/GetUniqueTagModesQueryHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using ServiceResult;
+
+namespace Equinor.Procosys.Preservation.Query.Tests.GetUniqueTagModes
+{
+    [TestClass]
+    public class GetUniqueTagModesQueryHandlerTests : ReadOnlyTestsBase
+    {
+        private TestDataSet _testDataSet;
+        private GetUniqueTagModesQuery _queryForProject1;
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Query.Tests/GetUniqueTagModes/GetUniqueTagModesQueryHandlerTests.cs]: rename to just _qyery or __queryForProject (some for similar test you made yesterday)

File: src/tests/Equinor.Procosys.Preservation.Query.Tests/GetUniqueTagModes/GetUniqueTagModesQueryHandlerTests.cs (hunk 1 above)
Comment (janid1967) [reply src/tests/Equinor.Procosys.Preservation.Query.Tests/GetUniqueTagModes/GetUniqueTagModesQueryHandlerTests.cs]: If You create sep tests on it, do the same in other similar tests

File: src/tests/Equinor.Procosys.Preservation.Query.Tests/GetUniqueTagModes/GetUniqueTagModesQueryHandlerTests.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        {
+            using (var context = new PreservationContext(_dbContextOptions, _plantProvider, _eventDispatcher, _currentUserProvider))
+            {
+                var dut = new GetUniqueTagModesQueryHandler(context);
+
+                var result = await dut.Handle(_queryForProject1, default);
+                Assert.AreEqual(1, result.Data.Count);
+                Assert.IsTrue(result.Data.Any(rt => rt.Title == _testDataSet.Mode1.Title));
+
+                result = await dut.Handle(new GetUniqueTagModesQuery(_testDataSet.Project2.Name), default);
```
Comment (TordJoranger) [inline src/tests/Equinor.Procosys.Preservation.Query.Tests/GetUniqueTagModes/GetUniqueTagModesQueryHandlerTests.cs]: _queryForProject2 variabel eller sådan

**Pull request merged**